src/flint/FlintTypes.jl

```diff
@@ -1008,8 +1008,8 @@ const flint_orderings = [:lex, :deglex, :degrevlex]
 
 @attributes mutable struct ZZMPolyRing <: MPolyRing{ZZRingElem}
   nvars::Int
-  nfields::Cint
-  ord::Int
+  nfields::Int
+  ord::Cint
```

Comment:
These field sizes were wrong; it didn't matter for us in practice but it showed up when I used `dump` to understand why `sort_terms!` didn't do what I thought it would...
```
julia> dump(polynomial_ring(QQ, [:u,:v]; internal_ordering = :degrevlex)[1]; maxdepth=1)
QQMPolyRing
  nvars: Int64 2
  nfields: Int32 3
  ord: Int64 4294967298
  deg: Int32 1
  rev: Int32 0
  lut: NTuple{64, Int64}
  lut1: NTuple{64, UInt8}
  S: Array{Symbol}((2,))
  __attrs: #undef
```